softlayer/vm/softlayer_vm.go

```diff
@@ -141,7 +139,23 @@ func (vm SoftLayerVM) SetMetadata(vmMetadata VMMetadata) error {
 }
 
 func (vm SoftLayerVM) ConfigureNetworks(networks Networks) error {
-	return NotSupportedError{}
+	virtualGuest, err := bslcommon.GetObjectDetailsOnVirtualGuest(vm.softLayerClient, vm.ID())
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Cannot get details from virtual guest with id: %d.", virtualGuest.Id)
+	}
+
+	oldAgentEnv, err := vm.agentEnvService.Fetch()
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Failed to unmarshal userdata from virutal guest with id: %d.", virtualGuest.Id)
+	}
+
+	oldAgentEnv.Networks = networks
+	err = vm.agentEnvService.Update(oldAgentEnv)
+	if err != nil {
+		return bosherr.WrapError(err, fmt.Sprintf("Configuring network setting on VirtualGuest with id: `%d`", virtualGuest.Id))
+	}
+
+	return nil
 }
 
 func (vm SoftLayerVM) AttachDisk(disk bslcdisk.Disk) error {
```

Comment:
Same deal here... we cannot be changing code without adding or changing tests. Sorry.